Add SIG Storage Spotlight blog article

fsmunoz opened the pull request:
Interview with SIG-Storage as part of the SIG-ContribEx comms series "Spotlight on..."

/cc @xing-yang

sftim commented:
/retitle Add SIG Storage Spotlight blog article

sftim commented on content/en/blog/2022/SIG-Storage Spotlight.md:
```diff
@@ -0,0 +1,86 @@
+---
+layout: blog
+title: "SIG-Storage Spotlight"
```
How about aiming for publication on the 2nd of August?
(we can move it earlier if editing is done by then).
```suggestion
title: "Spotlight on SIG Storage"
date: 2022-08-02
```

For the title, compare https://www.k8s.dev/blog/2022/02/04/sig-multicluster-spotlight-2022/

fsmunoz replied:
Publication date seems fine; I used https://kubernetes.io/blog/2020/06/30/sig-windows-spotlight-2020/ as a reference, hence the different approach.

sftim commented on content/en/blog/2022/SIG-Storage Spotlight.md:
```diff
@@ -0,0 +1,86 @@
+
+**XY**: In Kubernetes, there are multiple components involved for a volume operation. For example, creating a Pod to use a PVC has multiple components involved. There are the Attach Detach Controller and the external-attacher working on attaching the PVC to the pod. There’s the Kubelet that works on mounting the PVC to the pod. Of course the CSI driver is involved as well. There could be race con ...
+
+Another challenge is regarding core vs [Custom Resource Definitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) (CRD), not really storage specific. CRD is a great way to extend Kubernetes capabilities while not adding too much code to the Kubernetes core itself. However, this also means there are many external components that are needed when running a K ...
```
```suggestion
Another challenge is regarding core versus [CustomResourceDefinitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) (CRD), not really storage specific. CRD is a great way to extend Kubernetes capabilities while not adding too much code to the Kubernetes core itself. However, this also means there are many external components that are needed when running a Kubernetes cluster.
```

sftim commented on content/en/blog/2022/SIG-Storage Spotlight.md:
```diff
@@ -0,0 +1,86 @@
+
+Another challenge is regarding core vs [Custom Resource Definitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) (CRD), not really storage specific. CRD is a great way to extend Kubernetes capabilities while not adding too much code to the Kubernetes core itself. However, this also means there are many external components that are needed when running a K ...
+
+From the SIG Storage side, one most notable example is Volume Snapshot. Volume Snapshot APIs are defined as CRDs. API definitions and controllers are out-of-tree. There is a common snapshot controller and a snapshot validation webhook that should be deployed on the control plane, similar to how kube-controller-manager is deployed. Although Volume Snapshot is a CRD, it is a core feature of SIG Sto ...
```
```suggestion
From the SIG Storage side, one most notable example is VolumeSnapshot. Volume snapshot APIs are defined as CRDs. API definitions and controllers are out-of-tree. There is a common snapshot controller and a snapshot validation webhook that should be deployed on the control plane, similar to how `kube-controller-manager` is deployed. Although Volum Snapshot is a CRD, it is a core feature of SIG Storage.  It is recommended for the K8s cluster distros to deploy VolumeSnapshot CRDs, the snapshot controller, and the snapshot validation webhook, however, most of the time we _don’t_ see distros deploy them. So this becomes a problem for the storage vendors: now it becomes their responsibility to deploy these non-driver specific common components. This could cause conflicts if a customer wants to use more than one storage system and deploy more than one CSI driver.
```

sftim commented on content/en/blog/2022/SIG-Storage Spotlight.md:
```diff
@@ -0,0 +1,86 @@
+
+**FSM**: That’s where CSI enters the game?
+
+**XY**: Exactly, then there comes [Container Storage Interface](https://kubernetes-csi.github.io/docs/) (CSI). This is an industry standard trying to design common storage interfaces so that a storage vendor can write one plugin and have it work across a range of container orchestration systems (CO). Now Kubernetes is the main CO, but back when CSI just started, there were Docker, Mesos, Cloud Fo ...
```
```suggestion
**XY**: Exactly, then there comes [Container Storage Interface](https://kubernetes-csi.github.io/docs/) (CSI). This is an industry standard trying to design common storage interfaces so that a storage vendor can write one plugin and have it work across a range of container orchestration (CO) systems. Now Kubernetes is the main CO, but back when CSI just started, there were Docker, Mesos, Cloud Foundry, in addition to Kubernetes. CSI drivers are out-of-tree so bug fixes and releases can happen at their own pace.
```

sftim commented on content/en/blog/2022/SIG-Storage Spotlight.md:
```diff
@@ -0,0 +1,86 @@
+
+* [CSI Migration](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/625-csi-migration) is an on-going effort that SIG Storage has been working on for a few releases now. The goal is to move in-tree volume plugins to out-of-tree CSI drivers and eventually remove the in-tree volume plugins.  There are 7 KEPs that we are targeting in 1.25 are related to CSI migration. There is  ...
+* The second one I want to highlight is [COSI, the Container Object Storage Interface](https://github.com/kubernetes-sigs/container-object-storage-interface-spec). This is a sub-project under SIG Storage. COSI proposes object storage Kubernetes APIs to support orchestration of object store operations for Kubernetes workloads. It also introduces gRPC interfaces for object storage providers to writ ...
+* Another feature I want to mention is [CSI Ephemeral Volume](https://github.com/kubernetes/enhancements/issues/596) support. This feature allows CSI volumes to be specified directly in the pod specification for ephemeral use cases. They can be used to inject arbitrary states, such as configuration, secrets, identity, variables or similar information, directly inside pods using a mounted volume.  ...
```
How about hyperlinking to https://kubernetes.io/docs/concepts/storage/ephemeral-volumes/#csi-ephemeral-volumes instead of https://github.com/kubernetes/enhancements/issues/596?

A link to the KEP _as well_ could be OK; eg, the text “targeting GA” could link to https://github.com/kubernetes/enhancements/issues/596#issuecomment-1161257318

fsmunoz replied:
That's sensible, changed.

sftim commented on content/en/blog/2022/SIG-Storage Spotlight.md:
```diff
@@ -0,0 +1,86 @@
+title: "SIG-Storage Spotlight"
+slug: sig-storage-spotlight
+---
+
```
Add an author paragraph here.

sftim commented on content/en/blog/2022/SIG-Storage Spotlight.md:
```diff
@@ -0,0 +1,86 @@
+
+We have bi-weekly meetings where we go through features we are working on for each particular release, getting the statuses, making sure each feature has dev owners and reviewers working on it, and reminding people about the release deadlines, etc. More information on the SIG is on the [community page](https://github.com/kubernetes/community/tree/master/sig-storage). People can also add PRs that  ...
+
+We also have other regular meetings, i.e., CSI Implementation meeting, Object Bucket API design meeting, and one-off meetings for specific topics if needed. There is also a [K8s Data Protection Workgroup](https://github.com/kubernetes/community/blob/master/wg-data-protection/README.md) that is sponsored by SIG Storage and SIG Apps. SIG Storage owns or co-owns features that are being discussed at  ...
```
```suggestion
We also have other regular meetings, i.e., CSI Implementation meeting, Object Bucket API design meeting, and one-off meetings for specific topics if needed. There is also a K8s [Data Protection Working Group](https://github.com/kubernetes/community/blob/master/wg-data-protection/README.md) that is sponsored by SIG Storage and SIG Apps. SIG Storage owns or co-owns features that are being discussed at the Data Protection WG.
```

sftim commented on content/en/blog/2022/SIG-Storage Spotlight.md:
```diff
@@ -0,0 +1,86 @@
+
+**FSM**: Storage is such a foundational component in so many things, not least in Kubernetes: what do you think are the Kubernetes-specific challenges in terms of storage management?
+
+**XY**: In Kubernetes, there are multiple components involved for a volume operation. For example, creating a Pod to use a PVC has multiple components involved. There are the Attach Detach Controller and the external-attacher working on attaching the PVC to the pod. There’s the Kubelet that works on mounting the PVC to the pod. Of course the CSI driver is involved as well. There could be race con ...
```
nit: “kubelet” (all lower case) please.

sftim commented on content/en/blog/2022/SIG-Storage Spotlight.md:
```diff
@@ -0,0 +1,86 @@
+
+**XY**: Yes. For the 1.25 release, I want to highlight the following:
+
+* [CSI Migration](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/625-csi-migration) is an on-going effort that SIG Storage has been working on for a few releases now. The goal is to move in-tree volume plugins to out-of-tree CSI drivers and eventually remove the in-tree volume plugins.  There are 7 KEPs that we are targeting in 1.25 are related to CSI migration. There is  ...
```
How about:
```suggestion
* [CSI Migration](https://kubernetes.io/blog/2021/12/10/storage-in-tree-to-csi-migration-status-update/#quick-recap-what-is-csi-migration-and-why-migrate) is an on-going effort that SIG Storage has been working on for a few releases now. The goal is to move in-tree volume plugins to out-of-tree CSI drivers and eventually remove the in-tree volume plugins.  There are 7 KEPs that we are targeting in 1.25 are related to CSI migration. There is one [core KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/625-csi-migration) for the general CSI Migration feature. That is targeting GA in 1.25. CSI Migration for GCE PD and AWS EBS are targeting GA. CSI Migration for vSphere is targeting to have the feature gate on by default while staying in 1.25 that are in Beta. Ceph RBD and PortWorx are targeting Beta, with feature gate off by default. Ceph FS is targeting Alpha.
```

sftim reviewed (commented):
Thanks. Here's a review.

fsmunoz commented:
@sftim I've added those suggestions which weren't directly commitable in my last commit, I think it covers everything.

xing-yang commented on content/en/blog/2022/SIG-Storage Spotlight.md:
```diff
@@ -0,0 +1,88 @@
+
+Another challenge is regarding core versus [CustomResourceDefinitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) (CRD), not really storage specific. CRD is a great way to extend Kubernetes capabilities while not adding too much code to the Kubernetes core itself. However, this also means there are many external components that are needed when running a ...
+
+From the SIG Storage side, one most notable example is VolumeSnapshot. Volume snapshot APIs are defined as CRDs. API definitions and controllers are out-of-tree. There is a common snapshot controller and a snapshot validation webhook that should be deployed on the control plane, similar to how `kube-controller-manager` is deployed. Although Volum Snapshot is a CRD, it is a core feature of SIG Sto ...
```
typo: Volum Snapshot -> Volume Snapshot

xing-yang replied:
Can you remove "and a snapshot validation webhook" from this line?  I just realized that the webhook should be deployed together with the API server.

fsmunoz commented:
Thank you @xing-yang, I have reflected your review in my last commit.

sftim commented on content/en/blog/2022/SIG-Storage Spotlight.md:
```diff
@@ -0,0 +1,88 @@
+---
+layout: blog
+title: "Spotlight on SIG Storage"
+date: 2022-08-02
```
This is likely to follow close on the SIG Docs spotlight article. How about we defer this for a week or so (more if folks want to)?

fsmunoz replied:
Not a problem to me, whenever it's suitable.

sftim commented on content/en/blog/2022/SIG-Storage Spotlight.md:
```diff
@@ -0,0 +1,88 @@
+layout: blog
+title: "Spotlight on SIG Storage"
+date: 2022-08-02
+slug: sig-storage-spotlight
```
To match what seems to be a convention:
```suggestion
slug: sig-storage-spotlight-2022
```

sftim commented:
I just noticed the filename: `SIG-Storage Spotlight.md`. Could I get that changed to not contain a space, and be lowercase, ie: `content/en/blog/2022/sig-storage-spotlight.md`?

fsmunoz commented:
> I just noticed the filename: `SIG-Storage Spotlight.md`. Could I get that changed to not contain a space, and be lowercase, ie: `content/en/blog/2022/sig-storage-spotlight.md`?

Done in last commit 3d3e8b4

fsmunoz commented:
@sftim can you approve this? I was under the impression that only @mrbobbytables could (since that's what the bot says), but it seems that anyone in the OWNERS file can.

sftim commented:
Ideally, we'd publish this to this blog and to the main Kubernetes blog on the same day. If you'd like that @fsmunoz , please open a PR against k/website to mirror this; see https://github.com/kubernetes/website/pull/35530 for an example.

BTW the front matter and filename would be different between the 2 sites; the rest of the article can usually be an exact copy.

fsmunoz commented:
@sftim Thank you! I have updated the publication date in the last commit. I have also replaced the content of the initial PR I had opened against the main website. I can change the filename but I am not sure to what https://github.com/kubernetes/website/pull/35084

sftim commented:
/hold
due to issue https://github.com/kubernetes/contributor-site/issues/281

/lgtm
/approve

sftim commented:
/hold cancel
